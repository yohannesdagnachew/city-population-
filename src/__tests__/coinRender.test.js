import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import DetailsRender from '../coins/coinsDetails/DetailsRender';
import store from '../redux/configureStore';

it('To check test is working', () => {
  expect(true).toBe(true);
});

it('Test if coinrenderer render correctly', () => {
  const list = {
    id: 'bit',
    icon: 'http/image',
    name: 'coinbase',
    rank: 2,
    marketCap: 20378420,
    symbol: 'image/24',
    price: 200,
    volume: 68,
    availableSupply: 3233,
    totalSupply: 3232,
    priceChange1h: 545,
    priceChange1d: 34,
    priceChange1w: 64,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <DetailsRender List={list} />
    </Provider>
    ,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test if coinrenDetail render correctly', () => {
  const list = {
    id: 'bitcoin', icon: 'https//image1.com', price: 100, name: 'jo',
  };
  const tree = renderer.create(
    <Provider store={store}>
      <DetailsRender List={list} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
