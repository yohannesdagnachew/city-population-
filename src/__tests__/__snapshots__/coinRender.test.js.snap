// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if coinrenDetail render correctly 1`] = `
<div
  className="all-info"
>
  <div
    className="header-info"
    id="bitcoin"
  >
    <div>
      jo
    </div>
    <img
      alt="bitcoin"
      className="icons-detail"
      src="https//image1.com"
    />
    <div>
      Price : 
      $100.00
    </div>
  </div>
  <div
    className="info"
  >
    <div
      className="info-right"
    >
      <p>
        Rank : 
      </p>
      <p>
        Symbol : 
      </p>
      <p>
        Price : 
        $100.00
      </p>
      <p>
        Volume : 
      </p>
      <p>
        MarketCap : 
      </p>
    </div>
    <div
      className="info-left"
    >
      <p>
        Av.Supply : 
      </p>
      <p>
        Tot.Supply : 
      </p>
      <p>
        Price var/H : 
         
        % /hour
      </p>
      <p>
        Price var/D :
         
        % /day
      </p>
      <p>
        Price var/W :
         
        % /week
      </p>
    </div>
  </div>
</div>
`;

exports[`Test if coinrenderer render correctly 1`] = `
<div
  className="all-info"
>
  <div
    className="header-info"
    id="bit"
  >
    <div>
      coinbase
    </div>
    <img
      alt="bit"
      className="icons-detail"
      src="http/image"
    />
    <div>
      Price : 
      $200.00
    </div>
  </div>
  <div
    className="info"
  >
    <div
      className="info-right"
    >
      <p>
        Rank : 
        2
      </p>
      <p>
        Symbol : 
        image/24
      </p>
      <p>
        Price : 
        $200.00
      </p>
      <p>
        Volume : 
        68
      </p>
      <p>
        MarketCap : 
        20378420
      </p>
    </div>
    <div
      className="info-left"
    >
      <p>
        Av.Supply : 
        3233
      </p>
      <p>
        Tot.Supply : 
        3232
      </p>
      <p>
        Price var/H : 
         
        545
        % /hour
      </p>
      <p>
        Price var/D :
         
        34
        % /day
      </p>
      <p>
        Price var/W :
         
        64
        % /week
      </p>
    </div>
  </div>
</div>
`;
